packages/marketplace/src/components/ui/client-app-detail/client-app-install-confirmation.tsx

```diff
@@ -0,0 +1,167 @@
+    )
+  }
+}
+
+export const handleSuccessAlertButtonClick = history => {
+  return () => {
+    history.replace(routes.CLIENT)
+  }
+}
+
+export const handleSuccessAlertMessageAfterClose = (setIsSuccessAlertVisible: (isVisible: boolean) => void) => {
+  return () => {
+    setIsSuccessAlertVisible(false)
+  }
+}
+
+const ClientAppInstallConfirmation: React.FC<ClientAppInstallConfirmationProps> = ({
+  appDetailData,
+  visible,
+  closeInstallConfirmationModal,
+}) => {
+  const history = useHistory()
+  const [isSuccessAlertVisible, setIsSuccessAlertVisible] = React.useState(false)
+  const clientId = useSelector(selectClientId)
+  const installationFormState = useSelector(selectInstallationFormState)
+  const isSubmitting = installationFormState === 'SUBMITTING'
+
+  const { name, id = '', scopes = [] } = appDetailData || {}
+
+  const dispatch = useDispatch()
+  const onSuccessAlertButtonClick = React.useCallback(handleSuccessAlertButtonClick(history), [history])
+
+  return (
+    <>
+      <Modal
+        visible={visible}
+        title={`Confirm ${name} installation`}
+        afterClose={closeInstallConfirmationModal}
+        footerItems={
+          <>
+            <Button
+              dataTest="agree-btn"
+              loading={isSubmitting}
+              className={appPermissionContentStyles.installButton}
+              type="button"
+              variant="primary"
+              onClick={handleInstallButtonClick(
+                id,
+                clientId,
+                dispatch,
+                setIsSuccessAlertVisible,
+                closeInstallConfirmationModal,
+              )}
+            >
+              Confirm
+            </Button>
+            <Button
+              dataTest="disagree-btn"
+              disabled={false}
+              className={appPermissionContentStyles.installButton}
+              type="button"
+              variant="danger"
+              onClick={closeInstallConfirmationModal}
+            >
+              Cancel
+            </Button>
+          </>
+        }
+      >
+        <>
+          {scopes.length ? (
+            <div>
+              <p>
+                This action will install the app for ALL platform users.
+                <br />
+                {name} requires the permissions below. By installing you are granting permission to your data.
+              </p>
+              <ul className={appPermissionContentStyles.permissionList}>
+                {scopes.map(({ description, name }) => (
+                  <li key={name} className={appPermissionContentStyles.permissionListItem}>
+                    {description}
+                  </li>
+                ))}
+              </ul>
+            </div>
+          ) : (
+            <p>This action will install the app for ALL platform users.</p>
+          )}
+        </>
+      </Modal>
+      <Modal visible={isSuccessAlertVisible} afterClose={handleSuccessAlertMessageAfterClose(setIsSuccessAlertVisible)}>
```

Comment:
I think should un mount if we don't use modal `isSuccessAlertVisible && <Modal visible={isSuccessAlertVisible} ...`